### ISSUE TITLE
Bug: XLIFF changes source after changed translation

### ISSUE THREAD

**Issue by kevinpapst** [WeblateOrg/weblate#6687]
**Describe the issue**

A short while ago I started to get translated `<source>` tags in my project, which is definitely a bug, as this tag is the main identifier to load the translation when using XLIFF. It happened multiple times by now, so I assume this is more than just "an accident". 

See this screenshot:
<img width="1638" alt="Bildschirmfoto 2021-10-17 um 19 27 45" src="https://user-images.githubusercontent.com/533162/137638241-96a2a5d0-c454-441a-9917-8f78904ba57a.png">

And this PR:
https://github.com/kevinpapst/kimai2/pull/2850/files

The original source ID  `error.too_many_entries` should NOT be touched under any circumstance.

This change seems to be copied into all languages, which have a translation for the string, eg:
<img width="1620" alt="Bildschirmfoto 2021-10-17 um 19 30 32" src="https://user-images.githubusercontent.com/533162/137638326-ddb6e6d7-f81f-4f05-8e46-0645d5f45ebe.png">

The first commit that introduced the problem: https://github.com/kevinpapst/kimai2/pull/2850/commits/1f9f02e7303cabac440ff09fcf06e8fa9ff00f8f

I think it was introduced by this Weblate change: 
https://hosted.weblate.org/translate/kimai/messages/en/?checksum=9d55afaba34c4891

<img width="882" alt="Bildschirmfoto 2021-10-17 um 19 36 01" src="https://user-images.githubusercontent.com/533162/137638516-7240421a-d734-4909-b45a-157fbcf351d5.png">

**I already tried**

Waiting for a fix, but it still exists, so I thought I'd share it.

I recognized it the first time a couple of weeks, so I assume it happened the first time between 4-8 weeks ago. But I cannot exactly pin point it, as I do not check the webpage PRs after every push.

**To Reproduce the issue**

I don't know how to reproduce it, sorry. Probably it is enough to change the translation in the main language of the project?

**Expected behavior**

Do NOT touch the `<source>` in XLIFF projects ever.

**Comment by ilocit** [comment]
@kevinpapst you are using XLIFF in a way it is not supposed to be used.
It is a bi-lingual translation file format and <source> should store the source text/string and not the "ID".
Once the XLIFF is translated, <target> will contain the translation.

**Comment by kevinpapst** [comment]
@ilocit well, having default translations as lookup strings is extremely error prone and for a good reason often not recommended. But I guess you know a lot more about translations than I do 😁 I am a consumer of translations and did not read the specs. Seems when I started to adapt XLIFF the framework documentation was "rather incomplete" and now I am here... as the Weblate behavior suddenly changed and caused a lot of manual work for me.

To answer my own question with a quote and link https://docs.weblate.org/en/latest/formats.html#string-keys :

> Weblate identifies the units in the XLIFF file by resname attribute in case it is present and falls back to id (together with file tag if present).

Closing, seems `resname` is the way to go forward.
